aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/service/StorageService.java

```diff
@@ -32,6 +33,14 @@
  */
 public interface StorageService {
 
+    /**
+     * Generate pre-signed download URL for an object.
+     * @param serviceKey key to uniquely specify item to generate URL for
+     * @param expires Number of seconds before URL expires
+     * @return A pre-signed URL
+     */
+    URL getPresignedUrl(@NonNull String serviceKey, int expires);
```

Comment:
Should it be called `generatePresignedUrl` to make clear that the URL doesn’t already exist. Otherwise right now this looks like an accessory method to a cached variable

Reply:
The storage service isn't exposed to the customer anyways so I think it's okay. I gave it the `get` prefix to be more aligned with `getUrl()` public API.